models/Document/Editable/Link.php

```diff
@@ -171,7 +171,7 @@ public function frontend()
 
             $attribs = array_unique($attribs);
 
-            return '<a href="'.$url.'" '.implode(' ', $attribs).'>' . $prefix . ($noText ? '' : htmlspecialchars($disabledText ? $url : $this->data['text'])) . $suffix . '</a>';
+            return '<a href="'.$url.'" '.implode(' ', $attribs).'>' . $prefix . ($noText ? '' : htmlspecialchars($disabledText ? $url : ($this->data['text'] ?? $url))) . $suffix . '</a>';
```

Comment:
```suggestion
            $text = '';
            if (!$noText) {
                $text = htmlspecialchars($disabledText ? $url : $this->data['text']);
            }
            
            return '<a href="'.$url.'" '.implode(' ', $attribs).'>' . $prefix . $text . $suffix . '</a>';
```
more readable?

Reply:
No because in the case i try to fix, `$noText` and `$disabledText` are always false, but the `text˙ key is not present in the data, so the error will still be there.

Reply:
yes, now the suggestion covers your fix as well.

Reply:
Am i seeing it wrong, or github messes up the diff? It looks like there are 2 `$text` assignments line 176 and line 177.

Reply:
It was typo. fixed it now